core/templates/dev/head/domain/exploration/StateObjectFactory.js

```diff
@@ -19,36 +19,47 @@
 
 oppia.factory('StateObjectFactory', [
   'AnswerGroupObjectFactory', 'InteractionObjectFactory',
-  function(AnswerGroupObjectFactory, InteractionObjectFactory) {
-  var State = function(name, classifierModelId, contentBackendList,
-    interactionBackendDict, paramChangesBackendList) {
-    this.name = name;
-    this.classifierModelId = classifierModelId;
-    this.content = contentBackendList;
-    this.interaction = InteractionObjectFactory.create(interactionBackendDict);
-    this.paramChanges = paramChangesBackendList;
-  };
+  'ContentObjectFactory',
+  function(AnswerGroupObjectFactory, InteractionObjectFactory,
+    ContentObjectFactory) {
+    var State = function(name, classifierModelId, contentBackendList,
```

Comment:
Oh, for stuff that hasn't been converted yet, we can leave them as dicts in the main constructor. But tidying this up now means that we're always clear that the inputs to createFromBackendDict() are meant to be dicts, and the inputs to createNew() are meant to be objects.

Temporarily, though, we can have createNew(..., someBackendDict, ...) and Constructor(someBackendDict) until the relevant objects are created, at which point we'd change those to objects as well.

Does that make sense? I may have misunderstood your comment.